spec/ancestor_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+require 'ostruct'
+
+RSpec.describe Kitchen::Ancestor do
+
+  let(:element) { OpenStruct.new(short_type: 'page') }
+  let(:descendant) { OpenStruct.new(short_type: 'note')}
+  let(:ancestor) { described_class.new(element) }
+
+  it 'makes the type available' do
+    expect(ancestor.type).to eq element.short_type
+  end
+
+  it 'makes the element available' do
+    expect(ancestor.element).to eq element
+  end
+
+  describe '#increment_descendant_count' do
+    it 'adds 1 to the descendant count for the given type' do
+      expect { ancestor.increment_descendant_count(descendant.short_type) }.to change {ancestor.get_descendant_count(descendant.short_type)}.by(1)
+    end
+  end
+
+  describe '#decrement_descendant_count' do
+    it 'decreases the descendant count for the given type by 1' do
+      ancestor.increment_descendant_count(descendant.short_type)
+      expect { ancestor.decrement_descendant_count(descendant.short_type) }.to change {ancestor.get_descendant_count(descendant.short_type)}.by(-1)
+    end
+    it 'decreases the descendant count for the given type by -5' do
+      expect { ancestor.decrement_descendant_count(descendant.short_type, by: -5) }.to raise_error("An element cannot have negative descendants")
+    end
+  end
+
+  describe '#get_descendant_count' do
+    it 'returns the descendant count for the given type' do
+      ancestor.increment_descendant_count(descendant.short_type)
+      expect(ancestor.get_descendant_count(descendant.short_type)).to eq 1
+    end
+  end
+
+  describe '#clone' do
+    it 'makes a new ancestor around the same element with new counts' do
+      @ancestor_clone = ancestor.clone()
```

Comment:
* You don't need the `()` when there are no arguments; it is more Ruby-like to just say `ancestor.clone`
* Don't need the `@` -- use that for instance variables in classes where you need to use the variable across methods.
* I recommend incrementing the descendant count on the ancestor before you clone it, that way you'll know that the `0` you're seeing in the clone is a true reset of the counts and not just what also happens to be in the original.